front/components/agent_builder/observability/ObservabilityFilterSelector.tsx

```diff
@@ -0,0 +1,128 @@
+import type { AgentVersionMarker } from "@app/lib/api/assistant/observability/version_markers";
+import { useAgentVersionMarkers } from "@app/lib/swr/assistants";
+
+interface ObservabilityFilterSelectorProps {
+  agentConfigurationId: string;
+}
+
+function getVersionValue(versionMarker: AgentVersionMarker) {
+  return `v${versionMarker.version}: ${versionMarker.timestamp}`;
+}
+
+export function ObservabilityFilterSelector({
+  agentConfigurationId,
+}: ObservabilityFilterSelectorProps) {
+  const {
+    mode,
+    setMode,
+    period,
+    setPeriod,
+    selectedVersion,
+    setSelectedVersion,
+  } = useObservabilityContext();
+
+  const { owner } = useAgentBuilderContext();
+  const { versionMarkers, isVersionMarkersLoading } = useAgentVersionMarkers({
+    workspaceId: owner.sId,
+    agentConfigurationId,
+    days: period,
+    disabled: false,
+  });
+
+  // Default to latest version when entering version mode with available markers
+  useEffect(() => {
+    if (
+      mode === "version" &&
+      !selectedVersion &&
+      versionMarkers &&
+      versionMarkers.length > 0
+    ) {
+      const value = getVersionValue(versionMarkers[versionMarkers.length - 1]);
+      setSelectedVersion(value);
+    }
+  }, [mode, selectedVersion, versionMarkers, setSelectedVersion]);
+
+  return (
+    <div className="flex items-center gap-3">
+      <ButtonsSwitchList defaultValue={mode} size="xs">
+        <ButtonsSwitch
+          value="timeRange"
+          label="Time range"
+          onClick={() => setMode("timeRange")}
+        />
+        <ButtonsSwitch
+          value="version"
+          label="Version"
+          onClick={() => setMode("version")}
+        />
+      </ButtonsSwitchList>
+
+      {mode === "timeRange" ? (
+        <DropdownMenu>
+          <DropdownMenuTrigger asChild>
+            <Button
+              label={`${period} days`}
+              size="xs"
+              variant="outline"
+              isSelect
+            />
+          </DropdownMenuTrigger>
+          <DropdownMenuContent align="end">
+            {OBSERVABILITY_TIME_RANGE.map((p) => (
+              <DropdownMenuItem
+                key={p}
+                label={`${p} days`}
+                onClick={() => setPeriod(p)}
+              />
+            ))}
+          </DropdownMenuContent>
+        </DropdownMenu>
+      ) : (
+        <DropdownMenu>
+          <DropdownMenuTrigger asChild>
+            <Button
+              label={
+                selectedVersion
+                  ? selectedVersion
+                  : isVersionMarkersLoading
+                    ? "Loading"
+                    : "Select"
+              }
+              size="xs"
+              variant="outline"
+              isSelect
+            />
+          </DropdownMenuTrigger>
+          <DropdownMenuContent align="end">
+            <DropdownMenuLabel label="Last 30 days" />
+            {(versionMarkers ?? []).map((m) => {
```

Comment:
What do we show if there's no version? (which shouldn't never happen btw?)

Reply:
wait I'm confused, it can happen or cannot happen? 🤔

Reply:
I guess we should not show the version option itself if there is no version??